parser/macro.go

```diff
@@ -205,8 +205,8 @@ type ExprHelper interface {
 		step *exprpb.Expr,
 		result *exprpb.Expr) *exprpb.Expr
 
-	// Ident creates an identifier Expr value.
-	Ident(name string) *exprpb.Expr
+	// Ident creates an identifier Expr value or error for e.g. reserved identifiers.
+	Ident(name string) (*exprpb.Expr, error)
```

Comment:
I think it would be preferable to keep the error an implementation detail in this case. Report it internally so that it shows up in the issues list, but don't return it to the caller. Kind of like what you're doing below in `GlobalCall`

Reply:
It looks like I was thinking of the `parser.VisitIdentOrGlobalCall` function which can report errors and _should_ be called before ever reaching the `helper.newIdent` function. Moving the checking up before getting into macro territory should be sufficient, I believe.

Reply:
Mooted by move to visitor.

Reply:
Yes, things are much cleaner just doing the check in VisitIdentOrGlobalCall().